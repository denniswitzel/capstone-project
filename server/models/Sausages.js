import mongoose from 'mongoose'

const SausagesSchema = mongoose.Schema({
    category: String,
    title: String,
    ingredients: String,
    allergies: [String],
    image: String,
    rating: Array
})

export default mongoose.model('sausages', SausagesSchema, 'sausages')