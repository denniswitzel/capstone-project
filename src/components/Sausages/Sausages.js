import React from 'react'
import sausages from '../../mocks/sausages.json'
import Headline from '../Headline/Headline'
import Product from '../Products/Product'

export default function Sausages(){
    return(
        <>
        <Headline headline="Sausages" />
        {sausages.map((sausage, index) => (
        <Product key={index} title={sausage.title} image={sausage.image} />
        ))}
        </>
    )
}
