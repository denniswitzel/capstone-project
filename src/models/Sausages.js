import mongoose from 'mongoose'

const SausagesSchema = mongoose.Schema({
    category: String,
    title: String,
    ingredients: String,
    allergies: Array,
    image: String,
    rating: Array
})

export default mongoose.model('sausages', SausagesSchema, 'sausages')