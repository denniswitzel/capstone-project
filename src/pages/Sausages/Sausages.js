import React from 'react'
import sausages from '../../mocks/sausages.json'
import Product from '../../components/Products/Product'
import Headline from '../../components/Headline/Headline'

export default function Sausages(){
    return(
        <>
        <Headline headline="Sausages" />
        {sausages.map((sausage, index) => (
        <Product key={index} title={sausage.title} image={sausage.image} />
        ))}
        </>
    )
}
