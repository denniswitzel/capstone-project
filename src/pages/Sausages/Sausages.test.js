import React from 'react'
import renderer from 'react-test-renderer'
import Sausages from '../Sausages/Sausages'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

describe('Sausages', () => {
    const title = 'Garden Gourmet - Sensational Bratwurst'
    const image = 'https://raw.githubusercontent.com/denniswitzel/capstone-project/master/src/images/sausages/Garden-Gourmet-Sensational-Bratwurst.png'

    it('display title and image', () => {
        const { getByText } = render(<Sausages title={title} image={image}/>)
        expect(getByText(title)).toBeInTheDocument()
        expect(document.querySelector("img").getAttribute("src")).toBe(image)
    })

    it('renders correctly', () => {
        const tree = renderer.create(<Sausages />)
        expect(tree).toMatchSnapshot()
      })
  })