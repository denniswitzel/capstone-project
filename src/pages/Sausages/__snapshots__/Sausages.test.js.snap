// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sausages renders correctly 1`] = `
Array [
  <section
    className="sc-AxhCb djfBDV"
  >
    <svg
      className="sc-AxgMl kRQlAl"
      onClick={[Function]}
    >
      back-arrow.svg
    </svg>
    <h2
      className="sc-AxhUy gghUbh"
    >
      Sausages
    </h2>
  </section>,
  <div
    className="sc-AxjAm jTSiGe"
  >
    <h3
      className="sc-AxirZ FzsEO"
    >
      Garden Gourmet - Sensational Bratwurst
    </h3>
    <img
      alt="Garden Gourmet - Sensational Bratwurst"
      className="sc-AxiKw erJOzJ"
      src="https://raw.githubusercontent.com/denniswitzel/capstone-project/master/src/images/sausages/Garden-Gourmet-Sensational-Bratwurst.png"
    />
  </div>,
  <div
    className="sc-AxjAm jTSiGe"
  >
    <h3
      className="sc-AxirZ FzsEO"
    >
      Like meat - Like Bratwurst
    </h3>
    <img
      alt="Like meat - Like Bratwurst"
      className="sc-AxiKw erJOzJ"
      src="https://raw.githubusercontent.com/denniswitzel/capstone-project/master/src/images/sausages/Like-meat-Bratwurst.png"
    />
  </div>,
  <div
    className="sc-AxjAm jTSiGe"
  >
    <h3
      className="sc-AxirZ FzsEO"
    >
      Like meat - Like Curry Knacker
    </h3>
    <img
      alt="Like meat - Like Curry Knacker"
      className="sc-AxiKw erJOzJ"
      src="https://raw.githubusercontent.com/denniswitzel/capstone-project/master/src/images/sausages/Like-meat-Curry-Knacker.png"
    />
  </div>,
  <div
    className="sc-AxjAm jTSiGe"
  >
    <h3
      className="sc-AxirZ FzsEO"
    >
      Like meat - Like Schinken Bratwurst
    </h3>
    <img
      alt="Like meat - Like Schinken Bratwurst"
      className="sc-AxiKw erJOzJ"
      src="https://raw.githubusercontent.com/denniswitzel/capstone-project/master/src/images/sausages/Like-meat-Schinken-Bratwurst.png"
    />
  </div>,
  <div
    className="sc-AxjAm jTSiGe"
  >
    <h3
      className="sc-AxirZ FzsEO"
    >
      Wheaty - Vegane Frankenberger
    </h3>
    <img
      alt="Wheaty - Vegane Frankenberger"
      className="sc-AxiKw erJOzJ"
      src="https://raw.githubusercontent.com/denniswitzel/capstone-project/master/src/images/sausages/Wheaty-Vegane-Frankenberger.png"
    />
  </div>,
  <div
    className="sc-AxjAm jTSiGe"
  >
    <h3
      className="sc-AxirZ FzsEO"
    >
      Wheaty - Vegane Weenies
    </h3>
    <img
      alt="Wheaty - Vegane Weenies"
      className="sc-AxiKw erJOzJ"
      src="https://raw.githubusercontent.com/denniswitzel/capstone-project/master/src/images/sausages/Wheaty-Vegane-Weenies.png"
    />
  </div>,
  <div
    className="sc-AxjAm jTSiGe"
  >
    <h3
      className="sc-AxirZ FzsEO"
    >
      Wheaty - Vegane Weenies
    </h3>
    <img
      alt="Wheaty - Vegane Weenies"
      className="sc-AxiKw erJOzJ"
      src="https://raw.githubusercontent.com/denniswitzel/capstone-project/master/src/images/sausages/Wheaty-Vegane-Weenies.png"
    />
  </div>,
  <div
    className="sc-AxjAm jTSiGe"
  >
    <h3
      className="sc-AxirZ FzsEO"
    >
      Wheaty - Vegane Weiße
    </h3>
    <img
      alt="Wheaty - Vegane Weiße"
      className="sc-AxiKw erJOzJ"
      src="https://raw.githubusercontent.com/denniswitzel/capstone-project/master/src/images/sausages/Wheaty-Vegane-Weiße.png"
    />
  </div>,
]
`;
